internal/provider/resource_integration.go

```diff
@@ -220,37 +246,26 @@ func (r integrationResource) Read(ctx context.Context, request resource.ReadRequ
 
 func (r integrationResource) Update(ctx context.Context, request resource.UpdateRequest, response *resource.UpdateResponse) {
 	var plan integrationData
-	diags := request.Plan.Set(ctx, &plan)
```

Comment:
😇 change `Set` to `Get`.
now we don't need `request.Config` workaround anymore.